### PR TITLE
Replace unsupported characters by underscore in labels

### PR DESCRIPTION
**Pull request by vmercierfr**
# Objective

Replace unsupported characters by underscore in labels

# Why

We introduced the `rds_instance_tags` metric in 0.5.0. AWS tags may contain characters that are not supported by Prometheus labels.

For instance, AWS RDS instances deployed in AWS ACK contain the `services.k8s.aws/controller-version` tag, which makes the Prometheus RDS exporter crash.

# How

- Clear Prometheus labels that are generated from AWS tags

# Release plan

- [ ] Merge this PR
- [ ] Release 0.5.1

### CODE REVIEW

**Comment by vmercierfr** [comment]
#sre

File: internal/app/exporter/helper.go [hunk 1]
```diff
@@ -24,3 +26,13 @@ func getUniqTypeAndIdentifiers(instances map[string]rds.RdsInstanceMetrics) ([]s
...
+
+func ClearPrometheusLabel(str string) string {
+	// Prometheus metric names may contain ASCII letters, digits, underscores, and colons.
+	// It must match the regex [a-zA-Z_:][a-zA-Z0-9_:]*.
+	// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+	InvalidFirstLetterCharacters := regexp.MustCompile(`[^a-zA-Z_:]+`)
+	InvalidCharacters := regexp.MustCompile(`[^a-zA-Z0-9_:]+`)
+
+	return InvalidFirstLetterCharacters.ReplaceAllString(string(str[0]), "_") + InvalidCharacters.ReplaceAllString(str[1:], "_")
+}
```
Comment (dcupif) [inline internal/app/exporter/helper.go:38]:
```suggestion
type Label string

func (l Label) sanitize() string {
	// Prometheus metric names may contain ASCII letters, digits, underscores, and colons.
	// It must match the regex [a-zA-Z_:][a-zA-Z0-9_:]*.
	// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
	invalidFirstLetterCharacters := regexp.MustCompile(`[^a-zA-Z_:]+`)
	invalidCharacters := regexp.MustCompile(`[^a-zA-Z0-9_:]+`)

	return InvalidFirstLetterCharacters.ReplaceAllString(string(str[0]), "_") + InvalidCharacters.ReplaceAllString(str[1:], "_")
}
```

**Review by dcupif — APPROVED** [review]
lgtm with suggestions

File: internal/app/exporter/helper.go (hunk 1 above)
Comment (vmercierfr) [reply internal/app/exporter/helper.go:38]: Great suggestion, thanks!

File: internal/app/exporter/helper.go [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package exporter
```
[the remark this answers is not in the document]
Comment (vmercierfr) [reply internal/app/exporter/helper.go:1]: Good idea, I applied it

**Pull request merged**